File: ClientAppV2/src/app/shared/models/userParams.ts
import { User } from "./user";

export class UserParams {
  gender: string;
  specializationId = 0;
  minAge = 18;
  maxAge = 99;
  pageNumber = 1;
  pageSize = 6;
  orderBy = 'lastActive';
  search = '';

  constructor(user: User) {
    this.gender = user.gender === 'female' ? 'male' : 'female'
  }
}

File: DAL/Specifications/UserParams.cs
namespace DAL.Specifications;

public class UserParams : PaginationParams
{
    public string CurrentUsername { get; set; }
    public string Gender { get; set; }
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 100;
    public string OrderBy { get; set; } = "lastActive";
    private string _search = "";

    public string Search
    {
        get => _search;
        set => _search = value.ToLower();
    }
}